js/src/data/actions.js

```diff
@@ -862,6 +870,75 @@ export function* deleteAdsCampaign( id ) {
 	}
 }
 
+/**
+ * Creates an asset group under the given Google Ads campaign.
+ *
+ * @param {number} campaignId The ID of the campaign to be created the asset group.
+ * @yield {Object} The wp-data action with data payload.
+ * @throws { { message: string } } Will throw an error if the creation fails.
+ */
+export function* createCampaignAssetGroup( campaignId ) {
+	try {
+		const response = yield apiFetch( {
+			path: `${ API_NAMESPACE }/ads/campaigns/asset-groups`,
+			method: 'POST',
+			data: { campaign_id: campaignId },
+		} );
+
+		return {
+			type: TYPES.CREATE_CAMPAIGN_ASSET_GROUP,
+			campaignId,
+			assetGroup: {
+				...EMPTY_ASSET_ENTITY_GROUP,
+				id: response.id,
+			},
+		};
+	} catch ( error ) {
+		const message =
+			error.message ||
+			__(
+				'There was an error creating the assets of the campaign.',
+				'google-listings-and-ads'
+			);
+
+		yield handleFetchError( error, message );
+		throw error;
+	}
+}
+
+/**
+ * Updates the asset group of the Google Ads campaign.
+ *
+ * @param {number} assetGroupId The ID of the asset group to be updated.
+ * @param {AssetEntityGroupUpdateBody} body The body of the updating request.
+ * @yield {Object} The wp-data action with data payload.
+ * @throws { { message: string } } Will throw an error if the update fails.
+ */
+export function* updateCampaignAssetGroup( assetGroupId, body ) {
+	try {
+		yield apiFetch( {
+			path: `${ API_NAMESPACE }/ads/campaigns/asset-groups/${ assetGroupId }`,
+			method: 'PUT',
+			data: body,
+		} );
+
+		return {
+			type: TYPES.UPDATE_CAMPAIGN_ASSET_GROUP,
```

Comment:
We are using here `TYPES.UPDATE_CAMPAIGN_ASSET_GROUP,` but seems that is not implemented in the reducer?

Reply:
The API response doesn't have the updated asset group, so no mutations can be done in the reducer. I originally thought it might be good to still signal this update action to **wp-data** even if it changes nothing.

In addition, it could be used in the `shouldInvalidate` of `getCampaignAssetGroups` resolver if someday it needs to invalidate the state rather than calling `invalidateResolution` [here](https://github.com/woocommerce/google-listings-and-ads/blob/50bffc816f65f5b03fe13ae553b96be9c92a5bb6/js/src/pages/edit-paid-ads-campaign/index.js#L118), but it can be added when needed. Do you prefer removing this action type?

Reply:
not any preference. Seems like working. I just was curious why, yes it was confusing for me as you can see, I was looking for the `UPDATE_CAMPAIGN_ASSET_GROUP ` in reducer as we (almost) always do.  Just curious. In this case is it wrong to just return?

Reply:
> Just curious. In this case is it wrong to just return?

It will work well. Like the `requestPhoneVerificationCode`, although it's not just `return;`, the return doesn't contain the `type` property and it works well. Recuder ignores it as the `switch` is mismatched.

Reply:
After actual testing, the `yield` or `return` actions without `type` property won't be passed to the reducer.